src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -36,13 +38,28 @@ internal static class CollectCommandHandler
         /// <param name="clrevents">A list of CLR events to be emitted.</param>
         /// <param name="clreventlevel">The verbosity level of CLR events</param>
         /// <returns></returns>
-        private static async Task<int> Collect(CancellationToken ct, IConsole console, int processId, FileInfo output, uint buffersize, string providers, string profile, TraceFileFormat format, TimeSpan duration, string clrevents, string clreventlevel)
+        private static async Task<int> Collect(CancellationToken ct, IConsole console, int processId, FileInfo output, uint buffersize, string providers, string profile, TraceFileFormat format, TimeSpan duration, string clrevents, string clreventlevel, string name)
         {
             try
             {
                 Debug.Assert(output != null);
                 Debug.Assert(profile != null);
 
+                // Either processName or processId has to be specified.
+                if (name!= "")
+                {
+                    if (processId != 0)
+                    {
+                        Console.WriteLine("Can only specify either --name or --process-id option.");
+                        return ErrorCodes.ArgumentError;
+                    }
+                    processId = CommandUtils.FindProcessIdWithName(name);
+                    if (processId < 0)
+                    {
+                        return ErrorCodes.ArgumentError;
+                    }
+                }
+
```

Comment:
@jonsequitur is there any way to consolidate logic like this into a shared argument validator so we don't need to have snippets like this at the top of every collector that uses a "shared" argument?  e.g., logic that dictates: "a command may take _either_ a `pid` which is valid if X or a `name` which is valid if Y"?  @sywhang, if this is non-trivial, it's not worth doing in this PR.